internal/ingress/controller/store/store.go

```diff
@@ -1049,7 +1049,14 @@ func (s *k8sStore) GetService(key string) (*corev1.Service, error) {
 
 func (s *k8sStore) GetIngressClass(ing *networkingv1.Ingress, icConfig *ingressclass.Configuration) (string, error) {
 	// First we try ingressClassName
-	if !icConfig.IgnoreIngressClass && ing.Spec.IngressClassName != nil {
+	if ing.Spec.IngressClassName != nil {
+		if icConfig.IgnoreIngressClass {
+			if icConfig.AnnotationValue == *ing.Spec.IngressClassName {
+				return *ing.Spec.IngressClassName, nil
+			}
```

Comment:
so  you ignore IngressClassName for one thing but use it for another thing?

This is not right IMHO, you are changing the semantics of two options, one public API of Ingress and one config of ingress-nginx ... this is a -1 as it is a breaking change for two things

Reply:
==Before change, if IgnoreIngressClass is false and  ing.Spec.IngressClassName is not empty,  ingress-controller mandate the IngressClass validation. it mean fails directly if no read permission on IngressClass resource

==After change 
 if IgnoreIngressClass is false and  ing.Spec.IngressClassName is not empty, stay as before.  if IgnoreIngressClass is true && ( ing.Spec.IngressClassName == <ingress class defined by cli "--ingress-class" ), ignore the IngressClass resource matchness check

Reply:
How to set `IgnoreIngressClass` to true or false ?   Is there a helm chart related values-file key-value pair ?

Reply:
> How to set `IgnoreIngressClass` to true or false ? Is there a helm chart related values-file key-value pair ?

You can always pass it as `controller.extraArgs`. I never got why users would add new values to the chart for stuff that's just being passed directly in to the arguments. But that's not to be discussed here, just my 2 cents. 😅

> if IgnoreIngressClass is false and ing.Spec.IngressClassName is not empty, stay as before. if IgnoreIngressClass is true && ( ing.Spec.IngressClassName == <ingress class defined by cli "--ingress-class" ), ignore the IngressClass resource matchness check

And that's what @aojea means: `.spec.ingressClassName` and what is should be used for is defined by a public API. We are not going to break this API.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#ingressspec-v1-networking-k8s-io

> ingressClassName is the name of an IngressClass cluster resource. Ingress controller implementations use this field to know whether they should be serving this Ingress resource, by a transitive connection (controller -> IngressClass -> Ingress resource). Although the `kubernetes.io/ingress.class` annotation (simple constant name) was never formally defined, it was widely supported by Ingress controllers to create a direct binding between Ingress controller and Ingress resources. Newly created Ingress resources should prefer using the field. However, even though the annotation is officially deprecated, for backwards compatibility reasons, ingress controllers should still honor that annotation if present.